src/traccuracy/utils.py

```diff
@@ -136,3 +140,96 @@ def get_corrected_division_graphs_with_delta(
+
+    Raises:
+        ValueError: matched argument must be an instance of `Matched`
+        ValueError: results argument must be a list of `Results` objects
+        ValueError: Requested target frame buffer {target_frame_buffer} exceeds computed "
+            "frame buffer {max_frame_buffer}
+    """
+    if not isinstance(matched, Matched):
+        raise ValueError("matched argument must be an instance of `Matched`")
+
+    if not isinstance(results, list) or not all(isinstance(res, Results) for res in results):
+        raise ValueError("results argument must be a list of `Results` objects")
+
+    # Check if divs in results and frame buffer is valid
+    reannotate_div = False
+    for res in results:
+        if res.metric_info["name"] == "DivisionMetrics":
+            max_frame_buffer = res.metric_info["frame_buffer"]
+            if target_frame_buffer > max_frame_buffer:
+                raise ValueError(
+                    f"Requested target frame buffer {target_frame_buffer} exceeds computed "
+                    f"frame buffer {max_frame_buffer}"
+                )
+            else:
+                reannotate_div = True
+                relaxed = res.metric_info["relax_skips_gt"] or res.metric_info["relax_skips_pred"]
+
+    if reannotate_div:
+        gt, pred = get_corrected_division_graphs_with_delta(
+            matched, frame_buffer=target_frame_buffer, relax_skip_edges=relaxed
+        )
+    else:
+        gt = matched.gt_graph
+        pred = matched.pred_graph
+
+    # Determine names of geffs
+    gt_name = f"{gt.name}.geff" if gt.name else "gt.geff"
+    pred_name = f"{pred.name}.geff" if pred.name else "pred.geff"
+
+    # Write geffs
+    for tg, name in zip([gt, pred], [gt_name, pred_name], strict=True):
+        geff_path = os.path.join(out_zarr, name)
+        axis_names = [tg.frame_key]
+        if tg.location_keys is not None:
+            axis_names.extend(tg.location_keys)
+        write_nx(
+            graph=tg.graph,
+            store=geff_path,
+            axis_names=axis_names,
+            axis_types=["time"] + ["space"] * (len(axis_names) - 1),
+        )
+        # Update metadata for division flags with buffer
+        if reannotate_div:
+            meta = GeffMetadata.read(geff_path)
+            props_meta = {}
+            for flag in [
+                NodeFlag.TP_DIV,
+                NodeFlag.TP_DIV_SKIP,
+                NodeFlag.FP_DIV,
+                NodeFlag.FN_DIV,
+                NodeFlag.WC_DIV,
+            ]:
+                props_meta[str(flag)] = {
+                    "identifier": str(flag),
+                    "dtype": "bool",
+                    "description": f"Target frame buffer {target_frame_buffer}",
+                }
+            meta.node_props_metadata = props_meta
+            meta.write(geff_path)
+
+    # Write results json
+    with open(os.path.join(out_zarr, "traccuracy-results.json"), mode="w") as f:
```

Comment:
I think it would be nice to extract this one-liner into its own public util that takes a path, and then call that here. That way if someone just wants to write the results they can do that. Maybe instead of util it could be a results object method? Happy with either.